python/ray/dashboard/utils.py

```diff
@@ -358,6 +358,31 @@ def address_tuple(address):
     return ip, int(port)
 
 
+def node_stats_to_dict(message):
+    decode_keys = {
+        "actorId",
+        "jobId",
+        "taskId",
+        "parentTaskId",
+        "sourceActorId",
+        "callerId",
+        "rayletId",
+        "workerId",
+        "placementGroupId",
+    }
+    core_workers_stats = message.core_workers_stats
+    message.ClearField("core_workers_stats")
+    try:
+        result = message_to_dict(message, decode_keys)
+        result["coreWorkersStats"] = [
+            message_to_dict(m, decode_keys, always_print_fields_with_no_presence=True)
+            for m in core_workers_stats
+        ]
+        return result
+    finally:
+        message.core_workers_stats.extend(core_workers_stats)
```

Comment:
do you know why this is in a finally block? very strange pattern

Reply:
1. The function first clears the field `core_workers_stats` from the message.
2. It converts the message into a dict and constructs `result["coreWorkersStats"]`.
3. `finally`: if step (2) throws an exception, the field `core_workers_stats` is set back on the message to avoid side effects.

Reply:
Still don't understand the point of clearing the field at all though, unless `message_to_dict` modifies the passed message in-place for some reason (unlikely).

Reply:
I guess the original author of this function thinks message will be updated in-place.

Reply:
Let me do some verification.

Reply:
The following test passes with the following change, so you are correct. It's unnecessary.

```sh
pytest -vs test_advanced.py::test_internal_free_non_owned
```

<img width="642" alt="image" src="https://github.com/user-attachments/assets/9df8822d-530c-4469-8932-77a5911373c5" />

Reply:
I have already removed `finally` https://github.com/ray-project/ray/pull/51187/commits/bf198639f215265473f128ae451204dac10ab789